### PR TITLE
Spanned name for Var Definition

### PR DESCRIPTION
**Pull request by adjivas**
Short PR to prepare EnumDef check cases of issue #5 .

### CODE REVIEW

File: telamon-gen/src/ast/context.rs [hunk 1]
```diff
@@ -362,8 +394,8 @@ impl TypingContext {
         // Type-check the base.
         let kind = body.kind;
         let all_var_defs = untyped_vars.iter().chain(&body.iter_vars).cloned().collect();
-        let vars = untyped_vars.into_iter().map(|def| {
-            (def.name.clone(), var_map.decl_argument(&self.ir_desc, def))
+        let vars = untyped_vars.iter().map(|def| {
+            (def.name.clone(), var_map.decl_argument(&self.ir_desc, def.clone()))
```
Comment (ulysseB) [inline telamon-gen/src/ast/context.rs]: Why do you need to add a copy ?

Comment (adjivas) [reply telamon-gen/src/ast/context.rs]: Specifically, it's a regression, I will have to restore it.

**Comment by adjivas** [comment]
[constrain.rs](https://github.com/ulysseB/telamon/blob/dfe2979bcc0d4933d9316dcc3714a8e6626f7476/telamon-gen/src/ast/constrain.rs) and [constraint.rs](https://github.com/ulysseB/telamon/blob/dfe2979bcc0d4933d9316dcc3714a8e6626f7476/telamon-gen/src/ast/constraint.rs) was same, that was a error of copy, that why I have removed this doublon.

**Pull request merged**